### PR TITLE
fix 1178 transparent object overlap solid objects

### LINKED ISSUE #1178: transparent object overlap solid objects
## Expected Behavior
image

## Actual Behavior
lines are drawn over the sphere and transparent cube too. This is not grid from jscad, these are actual transparent lines.
image

## Specifications
I actually introduced this bug in #1060 as I thought wrongly transparent shapes do not need depth buffer.
without depth buffer multiple overlaping transparent objects look better, but it is not actual solution to transparency,
as it causes issues like this, and it is not a viable compromise.

  - Version:
  - Platform:
  - Environment: (all)

### PR DESCRIPTION
**Pull request by hrgdavor**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission pass tests?

this fixes #1178 by removing code I added previously for transparent entities

### CODE REVIEW

**Comment by platypii** [comment]
I'm trying to test this locally, but I'm not seeing the same thing you are with transparency. And when I checkout this branch, I'm not seeing any change in behavior? Can you give an example, and maybe screenshots of JSCAD (not your prototype) showing the difference?

I tried:

```js
const main = () => {
  const s1 = colorize([0.8, 0.8, 0, 0.4], cube({ }))
  const s3 = colorize([0, 0.1, 0.9, 0.6], sphere({ center: [1.2, 0.2, 1] }))
  return [s1, s3]
}
```

And got:
Screenshot at 2022-12-22 13-07-20

**Comment by hrgdavor** [comment]
ok, nice test. looks like more things needed fixing because I probably worked around the issues in my prototype.

current jscad
image

with fix
image

code:
```js

const jscad=require('@jscad/modeling')
const {colorize} = jscad.colors
const {sphere,cube} = jscad.primitives

const main = () => {
  const s1 = colorize([0.8, 0.8, 0, 0.5], cube({ }))
  const s3 = colorize([0, 0.1, 0.9], sphere({ center: [1.2, 0.2, 1] }))
  return [s1, s3]
}

module.exports = {main}
```

**Review by platypii — DISMISSED** [review]
Beautiful! This is a significant improvement in transparency quality! Thanks @hrgdavor 

Screenshot at 2022-12-22 14-04-27

**Review by platypii — APPROVED** [review]

File: packages/utils/regl-renderer/src/rendering/commands/drawMesh/index.js [hunk 1]
```diff
@@ -9,14 +9,14 @@ const drawMesh = (regl, params = { extras: {} }) => {
     geometry: undefined,
     color: meshColor
   }
-  const { geometry, dynamicCulling, useVertexColors, color, transparent } = Object.assign({}, defaults, params)
-
+  const { geometry, dynamicCulling, useVertexColors, color, visuals } = Object.assign({}, defaults, params)
```
Comment (z3dev) [inline packages/utils/regl-renderer/src/rendering/commands/drawMesh/index.js:12]: hmm... it seems that we never had a 'default' value for transparent. let's assign a proper 'default' for visuals. maybe {}?

**Review by z3dev — CHANGES_REQUESTED** [review]
@hrgdavor looks like a very nice fix. i just have one small suggestion. then this is ready to go.

**Review by z3dev — APPROVED** [review]
@hrgdavor thanks again for the nice fixes!

**Comment by hrgdavor** [comment]
np, I introduced this bug 😄 in #1060

**Pull request merged**